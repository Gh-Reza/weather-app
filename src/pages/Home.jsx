import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { getCountries } from '../redux/countries/countriesSlice';
import Country from '../components/country';
import { getCountryStates } from '../redux/countries/statesSlice';
import Search from '../components/Search';

const Home = () => {
  const dispatch = useDispatch();

  const handleClick = (country) => {
    dispatch(getCountryStates(country));
  };

  useEffect(() => {
    dispatch(getCountries());
  }, [dispatch]);

  const countries = useSelector((store) => store.countries.countries);
  return (
    <main className="container container-md-fluid">
      <Search />
      <div className="container-fluid country-container">
        <div className="row row-cols-2 row-cols-sm-3 row-cols-md-4">
          {countries.map((country, i) => (
            <Country
              key={country.name}
              name={country.name}
              flag={country.flag}
              capital={country.capital[0]}
              lat={country.latlng[0]}
              lng={country.latlng[1]}
              onClick={handleClick}
              index={i}
            />
          ))}
        </div>
      </div>
    </main>
  );
};

export default Home;
