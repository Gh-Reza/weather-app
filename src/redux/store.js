import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import countriesReducer from './countries/countriesSlice';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
}, applyMiddleware);

export default store;
