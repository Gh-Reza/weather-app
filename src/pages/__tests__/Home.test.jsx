import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import '@testing-library/jest-dom/extend-expect';
import Home from '../Home';

describe('Tests the home page', () => {
  const component = (
    <Provider store={store}>
      <Home />
    </Provider>
  );

  test('Match the snapshot', () => {
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });

  test('Should contain an image component', () => {
    render(component);
    const main = screen.getByRole('main');
    expect(main).toBeInTheDocument();
  });
});
