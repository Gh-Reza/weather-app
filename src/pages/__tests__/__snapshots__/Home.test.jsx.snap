// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the home page Match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="gen container container-md-fluid"
      >
        <div
          class="row row-cols-2 row-cols-sm-3 row-cols-md-4"
        />
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="gen container container-md-fluid"
    >
      <div
        class="row row-cols-2 row-cols-sm-3 row-cols-md-4"
      />
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
